crates/adapters/src/controller.rs

```diff
@@ -147,10 +149,19 @@ pub(crate) const MAX_API_CONNECTIONS: u64 = 100;
 
 pub(crate) type EndpointId = u64;
 
-/// Latency of checkpoint operations, in microseconds.
-static CHECKPOINT_LATENCY: ExponentialHistogram = ExponentialHistogram::new();
+/// Runtime of checkpoint operations, in microseconds, including time that the
+/// pipeline could continue executing while the checkpoint completed.
+static CHECKPOINT_RUNTIME: ExponentialHistogram = ExponentialHistogram::new();
+
+/// Sub-duration of [CHECKPOINT_RUNTIME] during which pipeline execution was
+/// blocked.
+static CHECKPOINT_DELAY: ExponentialHistogram = ExponentialHistogram::new();
 
 /// Amount of storage written during checkpoint operations, in megabytes.
+///
+/// These values are approximate because they include all storage writes while a
+/// checkpoint is being written, which means that they include writes from
+/// ongoing background merges.
```

Comment:
I think naming this with _SIZE inside would be clearer

Reply:
I renamed it CHECKPOINT_WRITTEN_MEGABYTES.